<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Preprocessingn</class>
 <widget class="qMRMLWidget" name="Preprocessingn">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1284</width>
    <height>1377</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="FilteringCollapsibleButton">
     <property name="text">
      <string>Filtering</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="filterInputLabel">
        <property name="text">
         <string>Input volume:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="filterInputSelector">
        <property name="toolTip">
         <string>Select the input volume.</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="noneEnabled">
         <bool>false</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string>inputVolume</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="filterOutputLabel">
        <property name="text">
         <string>Filtered volume:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="filterOutputSelector">
        <property name="toolTip">
         <string>Create/select a node to store the filtered volume.</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLScalarVolumeNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="hideChildNodeTypes">
         <stringlist notr="true"/>
        </property>
        <property name="baseName">
         <string>VesselnessFiltered</string>
        </property>
        <property name="noneEnabled">
         <bool>false</bool>
        </property>
        <property name="addEnabled">
         <bool>true</bool>
        </property>
        <property name="removeEnabled">
         <bool>true</bool>
        </property>
        <property name="editEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
        <property name="interactionNodeSingletonTag">
         <string notr="true"/>
        </property>
        <property name="noneDisplay">
         <string>None</string>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string>vesselnessVolume</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="minDiameterLabel">
        <property name="toolTip">
         <string/>
        </property>
        <property name="text">
         <string>Minimum diameter (mm):</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="ctkSliderWidget" name="minDiameterSlider">
        <property name="toolTip">
         <string>Tubular structures that have minimum this diameter will be enhanced.</string>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>25.000000000000000</double>
        </property>
        <property name="value">
         <double>0.900000000000000</double>
        </property>
        <property name="tracking">
         <bool>true</bool>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string>minVesselDiameter</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="maxDiameterLabel">
        <property name="text">
         <string>Maximum diameter (mm):</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="ctkSliderWidget" name="maxDiameterSlider">
        <property name="toolTip">
         <string>Tubular structures that have maximum this diameter will be enhanced.</string>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
        <property name="value">
         <double>5.000000000000000</double>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string>maxVesselDiameter</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="vesselSeedLabel">
        <property name="text">
         <string>Vessel contrast seed:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="qMRMLNodeComboBox" name="contrastSeedSelector">
          <property name="toolTip">
           <string>Select a point in one vessel. This point is used to calculate the contrast between the vessel and the rest of the image.</string>
          </property>
          <property name="nodeTypes">
           <stringlist notr="true">
            <string>vtkMRMLMarkupsFiducialNode</string>
           </stringlist>
          </property>
          <property name="hideChildNodeTypes">
           <stringlist notr="true"/>
          </property>
          <property name="baseName">
           <string>Seeds</string>
          </property>
          <property name="noneEnabled">
           <bool>false</bool>
          </property>
          <property name="editEnabled">
           <bool>true</bool>
          </property>
          <property name="renameEnabled">
           <bool>true</bool>
          </property>
          <property name="interactionNodeSingletonTag">
           <string notr="true"/>
          </property>
          <property name="SlicerParameterName" stdset="0">
           <string>contrastSeed</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="qSlicerMarkupsPlaceWidget" name="contrastMarkupsPlaceWidget"/>
        </item>
       </layout>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="ctkCollapsibleButton" name="advancedFilteringCollapsibleButton">
        <property name="text">
         <string>Advanced filtering options</string>
        </property>
        <property name="collapsed">
         <bool>false</bool>
        </property>
        <layout class="QFormLayout" name="formLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="alphaLabel">
           <property name="text">
            <string>Alpha:            </string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="ctkSliderWidget" name="alphaSlider">
           <property name="toolTip">
            <string>A higher value filters out more plate-like structures.</string>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="pageStep">
            <double>10.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="value">
            <double>0.300000000000000</double>
           </property>
           <property name="suffix">
            <string/>
           </property>
           <property name="SlicerParameterName" stdset="0">
            <string>vesselnessAlpha</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="betaLabel">
           <property name="text">
            <string>Beta:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="ctkSliderWidget" name="betaSlider">
           <property name="toolTip">
            <string>A higher value filters out more blob-like structures.</string>
           </property>
           <property name="decimals">
            <number>1</number>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="maximum">
            <double>500.000000000000000</double>
           </property>
           <property name="value">
            <double>500.000000000000000</double>
           </property>
           <property name="suffix">
            <string/>
           </property>
           <property name="SlicerParameterName" stdset="0">
            <string>vesselnessBeta</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="filterMethodLabel">
           <property name="text">
            <string>Step method:                </string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QRadioButton" name="equispacedRadio">
             <property name="text">
              <string>Equispaced</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="logarithmicRadio">
             <property name="text">
              <string>Logarithmic</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Discretization steps:       </string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="filterStepsSpinBox">
           <property name="SlicerParameterName" stdset="0">
            <string>filteringSteps</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="8" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_12">
        <item>
         <widget class="QPushButton" name="cropPreviewButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Crop volume preview</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="deleteCropButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Delete crop</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="filterApplyButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Run the filtering algorithm.</string>
          </property>
          <property name="text">
           <string>Apply filtering</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="5" column="0">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>10</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_3">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QPushButton" name="changeModuleButton">
     <property name="text">
      <string>Go to Artery Segmentation module</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::MinimumExpanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerMarkupsPlaceWidget</class>
   <extends>qSlicerWidget</extends>
   <header>qSlicerMarkupsPlaceWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>Preprocessingn</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>filterInputSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>784</y>
    </hint>
    <hint type="destinationlabel">
     <x>253</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Preprocessingn</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>filterOutputSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>321</x>
     <y>819</y>
    </hint>
    <hint type="destinationlabel">
     <x>364</x>
     <y>90</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Preprocessingn</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>contrastSeedSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>474</x>
     <y>774</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Preprocessingn</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>contrastMarkupsPlaceWidget</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1071</x>
     <y>751</y>
    </hint>
    <hint type="destinationlabel">
     <x>1182</x>
     <y>180</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>contrastSeedSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>contrastMarkupsPlaceWidget</receiver>
   <slot>setCurrentNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1072</x>
     <y>176</y>
    </hint>
    <hint type="destinationlabel">
     <x>1123</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
